packages/grafana-ui/src/components/Legend/LegendListItem.tsx

```diff
@@ -0,0 +1,75 @@
+import React from 'react';
+import { css, cx } from 'emotion';
+import { LegendSeriesIcon } from './LegendSeriesIcon';
+import { LegendItem } from './types';
+import { SeriesColorChangeHandler } from './types';
+import { LegendStatsList } from './LegendStatsList';
+import { useTheme } from '../../themes/ThemeContext';
+import { stylesFactory } from '../../themes';
+import { GrafanaTheme } from '@grafana/data';
+
+export interface Props {
+  item: LegendItem;
+  className?: string;
+  onLabelClick?: (item: LegendItem, event: React.MouseEvent<HTMLDivElement>) => void;
+  onSeriesColorChange?: SeriesColorChangeHandler;
+}
+
+export const LegendListItem: React.FunctionComponent<Props> = ({ item, onSeriesColorChange, onLabelClick }) => {
+  const theme = useTheme();
+  const styles = getStyles(theme);
+
+  return (
+    <div className={styles.itemWrapper}>
+      <LegendSeriesIcon
+        disabled={!onSeriesColorChange}
+        color={item.color}
+        onColorChange={color => {
+          if (onSeriesColorChange) {
+            onSeriesColorChange(item.label, color);
+          }
+        }}
+        yAxis={item.yAxis}
+      />
+      <div
+        onClick={event => {
+          if (onLabelClick) {
+            onLabelClick(item, event);
+          }
+        }}
+        className={cx(styles.label, item.disabled && styles.labelDisabled)}
+      >
+        {item.label}
+      </div>
+
+      {item.displayValues && <LegendStatsList stats={item.displayValues} />}
+    </div>
+  );
+};
+
+LegendListItem.displayName = 'LegendTableItem';
+
+const getStyles = stylesFactory((theme: GrafanaTheme) => {
```

Comment:
FYI `stylesFactory` is not necessary until you need more args than theme. For theme only case yo can use [useStyles hook](https://github.com/grafana/grafana/blob/master/contribute/style-guides/themes.md#usestyles-hook)

Reply:
not sure I understand the advantage to useStyles? seems identical to stylesFactory but less. flexible? Why do we have two methods for the same thing?

Reply:
Less code is I guess the advantage.